### ISSUE TITLE
Comments above eslint-disable pass no-abusive-eslint-disable

### ISSUE THREAD

**Issue by viveleroi** [sindresorhus/eslint-plugin-unicorn#2752]
I updated to eslint 9 so that included a jump from unicorn 56 -> 61. 

I was trying to figure out why an ignore for abusive deps was being reported as useless, as no error was triggering. It turns it was because we had an unrelated comment above the eslint-disable.

I'm guessing this was an intentional change but it caused a lot of confusion, especially as the docs page doesn't list that as a passing scenario.

```js
export default defineConfig(unicorn.configs.recommended)
```

```js
// Test <-- remove this, and the error appears again
/* eslint-disable */
```

If this wasn't intended, it's a bug, otherwise it's not documented well and should be clarified. I also didn't see any mention in the release notes for any version between our old one and this new one.
